### PR TITLE
CS: refine ruined egg corridor logic

### PR DESCRIPTION
**Pull request by Miepee**
I have no clue on damage numbers for sisters, so not sure if this is maybe toooo conservative. 
|damage |after 1.5x|
|---|---|
|25|37,5|
|20|30|
|15|22,5|
|8|12|

### CODE REVIEW

File: randovania/games/cave_story/logic_database/Ruined Egg Corridor.txt [hunk 1]
```diff
@@ -182,10 +188,20 @@ Extra - starting_script: <TRA0051:0094:0004:0004
...
-      Any of the following:
-          Can Kill Bosses
-          Missiles ≥ 31 and Missile Launcher and Kill Bosses with Missiles (Expert)
-          Missiles ≥ 14 and Super Missile Launcher and Kill Bosses with Missiles (Advanced)
+      All of the following:
+          Any of the following:
+              # Weapon Requirements
+              Can Kill Bosses
+              # Kill with Missiles
+              Missiles ≥ 31 and Missile Launcher and Kill Bosses with Missiles (Beginner)
```
Comment (duncathan) [inline randovania/games/cave_story/logic_database/Ruined Egg Corridor.txt:196]: this was expert before

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added: Command line arguments for exporting games. These commands are intended for advanced uses only.
 - Fixed: During generation, actions that involves multiple progressive pickups are now properly considered.
 
+### Cave Story
+
+#### Logic Database
+
+##### Ruined Egg Corridor
+
+- Added: Health requirements to the Dragon Sisters.
```
Comment (duncathan) [inline CHANGELOG.md:19]: the name of the boss is just "the Sisters"

File: CHANGELOG.md [hunk 3]
```diff
@@ -10,6 +10,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
+- Fixed: Breaking the blocks in `Cthulu's Abode?` now properly accounts for Super Missile Launcher
```
Comment (duncathan) [inline CHANGELOG.md:20]: Cthulhu

**Review by duncathan — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (Miepee) [reply CHANGELOG.md]: Initially wanted to make it clearer that its the boss. But not like there's anything else in ruined egg thats also named `sisters` and makes sense within the health requirement context.

Comment (duncathan) [reply CHANGELOG.md]: I think it's much more clear if we call the boss by its name instead of something that isn't its name lol

**Review by duncathan — APPROVED** [review]

**Pull request merged**